### DISCUSSION TITLE
Question about event table (database)

### DISCUSSION THREAD

**Discussion by xKugeki** [dani-garcia/vaultwarden#4727] [Q&A]
Hey I was wondering if the data in 'event' is equal to the events that get logged in the web ui > organisation > reports ?

If I fail my login for about 20 times I can see that in the Report:
<img width="369" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/119607113/7a7cb1ae-bb8f-4d21-a7d8-477bd1b89db2">

But if I check the 'event' table I get only 5 entrys from today but none of them is logged with my ID or the eventID 1005
<img width="60" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/119607113/97e308ea-beb9-4a77-9f02-5b93df7128aa">
<img width="37" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/119607113/05dbf00e-a50a-49a4-9388-57f61e1572df">

I used the EventIDS from: https://bitwarden.com/help/event-logs/

Wish you a great day :D

**Comment by BlackDex** [comment]
The data most come from somewhere, and we do not magically generate data.
All is stored in the database and read from that same database.

If you are using sqlite then it might be that you only copied/used the actual db.sqlite3 file, and not used the wal/shm files. Those contain pending transactions to be written to the actual database, which might not have been handled yet.

**Reply by xKugeki** [reply]
I dont use sqlite I am currently using mysql. But how long does it takes the transactions to be written in the database? Even yesterdays entrys arent there.
Maybe I am to impatient :D then sorry about that.

But thank you very much for answering!

**Reply by BlackDex** [reply]
If it is MySQL then it should be instantly. We write it to the database, and to show it, we read it from the database.
The data isn't coming from anywhere else like memory or something else. If you can see it via the web-vault, it should be in the database.

**Reply by xKugeki** [reply]
mhm i tried exporting the organisation vault and that is logged in the reports but not in the event table. Maybe its just me having this issue. There is a missmatch with the time in the 'event' table and the server time. Maybe that could cause the problem?

**Reply by BlackDex** [reply]
I don't know. All i know is that what you see in the admin events is in the database, it can't be anywhere else.

All date time is stores as UTC.

**Reply by xKugeki** [reply]
Alright i tried that again with a new vaultwarden instance and there is everything up to date in the event table. Maybe it is because of EVENT_DAYS_RETAIN = 365 ? Maybe the table is full xD ?

**Reply by BlackDex** [reply]
The table can't be full unless you run out of disk space on your server. The retain is only used by the cleanup job which will remove older entries. It will not prevent new entries from being created.

Also, you do see them in the web-vault overview. And again, we can only show data which is located in that table, it doesn't come from anywhere else.
